### PR TITLE
Remove unnecessary shared_ptr for geology and black oil properties during redistribution.

### PR DESCRIPTION
**Pull request by blattms**
Since the refactoring of @dr-robertk there is no need for this anymore.

### CODE REVIEW

File: opm/autodiff/RedistributeDataHandles.hpp [hunk 1]
```diff
@@ -327,7 +325,8 @@ void distributeGridAndData( Dune::CpGrid& grid,
     grid.loadBalance(eclipseState);
     grid.switchToDistributedView();
 
-    distributed_props = std::make_shared<BlackoilPropsAdFromDeck>(properties, grid.numCells());
+    BlackoilPropsAdFromDeck distributed_props = BlackoilPropsAdFromDeck(properties,
+                                                                        grid.numCells());
     distributed_state.init(grid.numCells(), grid.numFaces(), state.numPhases());
```
Comment (bska) [inline opm/autodiff/RedistributeDataHandles.hpp]: Why not simply
```C++
BlackoilPropsAdFromDeck distributed_props(properties, grid.numCells());
```
or, for those who subscribe to "almost always `auto`",
```C++
auto distributed_props = BlackoilPropsAdFromDeck(properties, grid.numCells());
```

Comment (atgeirr) [inline opm/autodiff/RedistributeDataHandles.hpp]: I agree (and prefer the first one), but it is a minor detail. Will you fix it before I merge?

Comment (blattms) [inline opm/autodiff/RedistributeDataHandles.hpp]: handled with 4218a65

Comment (bska) [inline opm/autodiff/RedistributeDataHandles.hpp]: > handled with  4218a65 

Thanks.  Could you do the same transformation to `distributed_geology` on lines [341-343](https://github.com/blattms/opm-autodiff/blob/removed-superfluous-ptr-during-redistribution/opm/autodiff/RedistributeDataHandles.hpp#L341-L343)?  In other words, replace

```C++
DerivedGeology distributed_geology = DerivedGeology(/* args */);
```
by
```C++
DerivedGeology distributed_geology(/* args */);
```
?

Comment (blattms) [inline opm/autodiff/RedistributeDataHandles.hpp]: Sorry for the oversight and thanks for the hint. Proposed change applied.

Comment (bska) [inline opm/autodiff/RedistributeDataHandles.hpp]: > Sorry for the oversight and thanks for the hint. Proposed change applied.

Thanks a lot!

**Comment by bska** [comment]
Do you want to perform any additional work on the commits themselves or do you consider this to be complete now?

**Comment by blattms** [comment]
If it is OK, then I would squash/fixup all the commits into one.

**Comment by bska** [comment]
> If it is OK, then I would squash/fixup all the commits into one.

Go ahead.  I'll merge once you're done.

**Comment by blattms** [comment]
Fixed up everything into one commit. Please, go ahead and merge.

**Comment by bska** [comment]
> Fixed up everything into one commit. Please, go ahead and merge.

Thank you.  I'll merge into master right away

**Comment by atgeirr** [comment]
Just a note: unless specifically asked to, I'll assume that most things that go into the master branch is not required to also go into the release branch.

**Pull request merged**